deep_privacy/inference/infer.py

```diff
@@ -98,22 +98,32 @@ def pre_process(im, keypoint, bbox, imsize, cuda=True):
     return torch_input, keypoint, expanded_bbox, new_bbox
 
 
-def stitch_face(im, expanded_bbox, generated_face, bbox_to_extract, image_mask, original_bbox):
+def stitch_face(im, expanded_bbox, generated_face, bbox_to_extract, image_mask, original_bbox, tight_stitch=False):
     # Ugly but works....
+    # if tight_stitch is set to true, only the part of the image inside original_bbox is updated
     x0e, y0e, x1e, y1e = expanded_bbox
     x0o, y0o, x1o, y1o = bbox_to_extract
+    x0, y0, x1, y1 = original_bbox
+
+    if tight_stitch:
+        original_bbox_mask =  np.zeros_like(image_mask, dtype=bool)
+        original_bbox_mask[y0:y1, x0:x1, :] = 1
+        original_bbox_mask = original_bbox_mask * image_mask # logical and on masks
```

Comment:
I believe using [`np.logical_and`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.logical_and.html) is easier to read and we can remove the comment